### PR TITLE
Feature/404751 check personaldetails changes

### PR DESCRIPTION
**Pull request by MasilEviden**
Completed journey CheckYourDetails

### CODE REVIEW

File: src/FrontendAccountManagement.Web/Controllers/AccountManagement/AccountManagementController.cs [hunk 1]
```diff
@@ -299,6 +304,93 @@ public async Task<IActionResult> RemoveTeamMemberConfirmation(RemoveTeamMemberCo
...
+        return RedirectToAction("CheckYourDetails", editUserDetailsViewModel);
+    }
+    [HttpGet]
+    [Route(PagePath.CheckYourDetails)]
+    public async Task<IActionResult> CheckYourDetails( )
+    {
+        var session = await _sessionManager.GetSessionAsync(HttpContext.Session);
+        var userData = User.GetUserData();
+
+        SetBackLink(session, PagePath.CheckYourDetails);
```
Comment (NikPeters-Atos) [inline src/FrontendAccountManagement.Web/Controllers/AccountManagement/AccountManagementController.cs:362]: Does the back link work correctly? I found out there's some extra code to add when using this function to make it work properly

File: src/FrontendAccountManagement.Web/Controllers/AccountManagement/AccountManagementController.cs [hunk 2]
```diff
@@ -299,6 +304,93 @@ public async Task<IActionResult> RemoveTeamMemberConfirmation(RemoveTeamMemberCo
...
+        var model = new EditUserDetailsViewModel
```
Comment (NikPeters-Atos) [inline src/FrontendAccountManagement.Web/Controllers/AccountManagement/AccountManagementController.cs:363]: I would use the AutoMapper mapping function here

**Review by tim-powell-eviden — APPROVED** [review]

File: src/FrontendAccountManagement.Web/Controllers/AccountManagement/AccountManagementController.cs [hunk 3]
```diff
@@ -405,24 +406,33 @@ public async Task<IActionResult> EditUserDetails(EditUserDetailsViewModel editUs
...
         var session = await _sessionManager.GetSessionAsync(HttpContext.Session);
         var userData = User.GetUserData();
 
+        var editUserDetailsViewModel = new EditUserDetailsViewModel();
+
+        try
+        {
+            editUserDetailsViewModel = System.Text.Json.JsonSerializer.Deserialize<EditUserDetailsViewModel>(TempData["NewUserDetails"] as string);
+        }
+        catch (Exception) { }
```
Comment (NikPeters-Atos) [inline src/FrontendAccountManagement.Web/Controllers/AccountManagement/AccountManagementController.cs:426]: We shouldn't have an empty catch - what happens when this gets executed. IF it is right, and works like this, there should at least be a comment as to why.

File: src/FrontendAccountManagement.Web/Controllers/AccountManagement/AccountManagementController.cs [hunk 4]
```diff
@@ -405,24 +406,33 @@ public async Task<IActionResult> EditUserDetails(EditUserDetailsViewModel editUs
...
         var model = new EditUserDetailsViewModel
         {
-            FirstName = userData.FirstName,
-            LastName = userData.LastName,
-            JobTitle = userData.JobTitle,
-            Telephone = userData.Telephone
+            FirstName = editUserDetailsViewModel.FirstName ?? userData.FirstName,
+            LastName = editUserDetailsViewModel.LastName ?? userData.LastName,
+            JobTitle = editUserDetailsViewModel.JobTitle ?? userData.JobTitle,
+            Telephone = editUserDetailsViewModel.Telephone ??  userData.Telephone
```
Comment (NikPeters-Atos) [inline src/FrontendAccountManagement.Web/Controllers/AccountManagement/AccountManagementController.cs:435]: We should really be using AutoMapper here

File: src/FrontendAccountManagement.Web/Controllers/AccountManagement/AccountManagementController.cs [hunk 5]
```diff
@@ -437,13 +447,17 @@ public async Task<IActionResult> CheckYourDetails(EditUserDetailsViewModel model
         var session = await _sessionManager.GetSessionAsync(HttpContext.Session);
         var userData = User.GetUserData();
 
-        if (!ModelState.IsValid)
+        string servicerole = userData.ServiceRole ?? string.Empty;
+
+        if (servicerole.ToLower() == "admin")
```
Comment (NikPeters-Atos) [inline src/FrontendAccountManagement.Web/Controllers/AccountManagement/AccountManagementController.cs:452]: I believe there's already functionality to check service roles - it's an extension method on the ClaimsPrincipal

File: src/FrontendAccountManagement.Web/ViewModels/AccountManagement/EditUserDetailsViewModel.cs [hunk 6]
```diff
@@ -15,6 +15,9 @@ public class EditUserDetailsViewModel
         [StringLength(50, ErrorMessageResourceName = "LastNameMaxLength", ErrorMessageResourceType = typeof(EditUserDetails))]
         public string LastName { get; set; }
 
+        public string? OriginalFirstName { get; set; }
```
Comment (NikPeters-Atos) [inline src/FrontendAccountManagement.Web/ViewModels/AccountManagement/EditUserDetailsViewModel.cs:18]: This isn't needed - a user will always have a firstname and lastname

**Pull request merged**